internal/querynode/shard_cluster.go

```diff
@@ -303,6 +308,17 @@ func (sc *ShardCluster) transferSegment(old *shardSegmentInfo, evt segmentEvent)
 			sc.healthCheck()
 		}
 	case segmentStateLoaded:
+		// load balance
+		if old.nodeID != evt.nodeID {
+			sc.legacySegments = append(sc.legacySegments, shardSegmentInfo{
+				nodeID:    old.nodeID,
+				segmentID: old.segmentID,
+				inUse:     old.inUse,
+				state:     old.state,
+			})
+			// set segment inUse to zero, since there is no allocation for new segment
```

Comment:
I think the segment here refers to the 'old' segment, whose inUse counter is moved to the legacy segment.